code/src/mlx90640.cpp

```diff
@@ -0,0 +1,64 @@
+#include <mlx90640.hpp>
+
+namespace r2d2::thermal_camera {
+
+    mlx90640_c::mlx90640_c(i2c::i2c_bus_c &bus, const uint8_t address)
+        : mlx_i2c_bus(bus, address), mlx_processor(mlx_i2c_bus) {
+    }
+
+    void mlx90640_c::toggle_nth_bit(uint16_t &source, const uint8_t n,
+                                    const uint8_t to) const {
+        source = (source & ~(1U << n)) | (to << n);
+    }
+
+    uint16_t mlx90640_c::get_refresh_rate() const {
+        uint16_t rate = mlx_i2c_bus.read_register(INTERNAL_CONTROL_REGISTER);
+        rate >>= 7 & 7;
```

Comment:
You don't need the `&`, so `rate >>= 7;`

Reply:
When you have a bitset of 16 for example `1101 1101 0010 1101` and I shift this number to the right, 7 spots, that'll give me: `0000 0001 1011 1010`. If I don't `&` this with `7` aka `0000 0000 0000 0111`, this wouldn't give me the first three bits.

Reply:
Relying on the order of operations is unclear, could you change this to `rate = (rate >> 7) & 7`?